### PR TITLE
client-go: update client-go

### LINKED ISSUE #41758: Flashback fails due to "invalid request type StoreSafeTS" error
## Bug Report

Flashback fails after update client-go dependency. 

After setting log level to Warn for line 571, log show following error:

    [2023/02/27 16:31:34.234 +08:00] [WARN] [kv.go:554] ["update safeTS failed"] [error="invalid request type StoreSafeTS

https://github.com/tikv/client-go/blob/891ef575be7510892b16f40d553ce3b2442a95c9/tikv/kv.go#L569-L573

It causes TiDB fails to update store safe ts, and leads to flashback failure.

This bug is likely cause by https://github.com/tikv/client-go/pull/682

### 1. Minimal reproduce step (Required)

See https://github.com/tikv/tikv/issues/14282

### 2. What did you expect to see? (Required)

Flashback success.

### 3. What did you see instead (Required)

ERROR 1105 (HY000): cannot set flashback timestamp after min-resolved-ts(1970-01-01 00:00:00 +0000 UTC)

### 4. What is your TiDB version? (Required)

master branch @ a47f804

### LINKED ISSUE #41772: information_schema.DATA_LOCK_WAITS don't include global lock info
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

tiup deploy cluster
2 tidb
1 pd
1 tikv

```
# tidb-0-session
use test;
drop table if exists t;
create table t (a int primary key, b int, c varchar(255));
insert into t values(1,10,'a'),(2,20,'b'),(3,30,'c');
begin;
update t set b =31 where a = 3;

# tidb-1-session
begin;
update t set b=32 where a =3;

# tidb-0-session
select `key`, key_info, trx_id, current_holding_trx_id, sql_digest,sql_digest_text from information_schema.DATA_LOCK_WAITS;
```

### 2. What did you expect to see? (Required)
```
MySQL [test]> select `key`, key_info, trx_id, current_holding_trx_id, sql_digest,sql_digest_text from information_schema.DATA_LOCK_WAITS;
+----------------------------------------+----------------------------------------------------------------------------------------------------+--------------------+------------------------+------------+-----------------+
| key                                    | key_info                                                                                           | trx_id             | current_holding_trx_id | sql_digest | sql_digest_text |
+----------------------------------------+----------------------------------------------------------------------------------------------------+--------------------+------------------------+------------+-----------------+
| 7480000000000000575F728000000000000003 | {"db_name":"test","table_name":"t","handle_type":"int","handle_value":"3","db_id":2,"table_id":87} | 439761070133346307 |     439761048336072709 | NULL       | NULL            |
+----------------------------------------+----------------------------------------------------------------------------------------------------+--------------------+------------------------+------------+-----------------+
1 row in set (0.01 sec)
```
### 3. What did you see instead (Required)
```
MySQL [test]> select `key`, key_info, trx_id, current_holding_trx_id, sql_digest,sql_digest_text from information_schema.DATA_LOCK_WAITS;
Empty set (0.00 sec)
```

### 4. What is your TiDB version? (Required)

```
| Release Version: v6.7.0-alpha
Edition: Community
Git Commit Hash: cccada395af6d2e5778ebfbdf211c03c9a61a939
Git Branch: heads/refs/tags/v6.7.0-alpha
UTC Build Time: 2023-02-27 14:35:34
GoVersion: go1.20.1
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv |
```

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #41758, close #41772

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -92,7 +92,7 @@ require (
 	github.com/stretchr/testify v1.8.1
 	github.com/tdakkota/asciicheck v0.1.1
 	github.com/tiancaiamao/appdash v0.0.0-20181126055449-889f96f722a2
-	github.com/tikv/client-go/v2 v2.0.6-0.20230227032358-40a82457ebaa
```
Comment (hawkingrei) [inline go.mod:95]: Remember to run ```make bazel_preapre```.

Comment (Defined2014) [reply go.mod:95]: Done

**Review by hawkingrei — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**